### PR TITLE
DOC: add a docstring for the function 'compare_chararrays' (See #10106)

### PR DESCRIPTION
**Pull request by ryzheboka**
That's my first contribution, so I feel quit unsure.

### CODE REVIEW

**Comment by mattip** [comment]
I wonder why that function is public in the first place. Do you have a reason for using `compare_chararrays(a, b, <op>)` rather than using `a <op> b`?

**Comment by tylerjereddy** [comment]
I'm actually surprised you can do the operations without that function -- the string / unicode operations often seem to require their own functions for simple operations instead of assuming that i.e., `+` and so on will "just work."

**Comment by mattip** [comment]
This is specifically for the [`chararray` class](http://www.numpy.org/devdocs/reference/generated/numpy.chararray.html) which is a (deprecated?) subclass of `ndarray`, and the function is used in the implementation of the equivalent dunder operations

File: numpy/core/_add_newdocs.py [hunk 1]
```diff
@@ -1072,6 +1072,47 @@ def luf(lamdaexpr, *args, **kwargs):
...
+
+    TypeError
+        If at least one of the arrays a or b is a non-string array
+
+    Examples
+    --------
+    >>> a = np.array([["a"], ["b"], ["cde"]])
+    >>> b = np.array([["a"], ["a"], ["dec"]])
+    >>> np.compare_chararrays(a,b,">",True)
+    array([[False],[True],[False]])
```
[the remark this answers is not in the document]
Comment (ryzheboka) [reply numpy/core/_add_newdocs.py]: Thank you, I hadn't thought about that, so just changed it.

**Comment by charris** [comment]
@mattip It looks like the name is outdated and misleading, it works for string arrays in general.

**Comment by charris** [comment]
Thanks @ryzheboka .

**Pull request merged**